src/processors/universal.ts

```diff
@@ -370,6 +372,162 @@ class UniversalProcessor extends BaseOrderProcessor {
     return unlockTx;
   }
 
+  private unlockOrder(
+    orderId: string,
+    order: OrderData,
+    context: OrderProcessorContext
+  ) {
+    const giveChain = order.give.chainId;
+    // filling batch queue
+    let orderIds = this.unlockBatchesOrderIdMap.get(giveChain);
+    if (!orderIds) {
+      orderIds = new Set();
+    }
+    orderIds.add(orderId);
+    this.unlockBatchesOrderIdMap.set(giveChain, orderIds);
+    this.ordersDataMap.set(orderId, order);
+
+    // check that process is blocked
+    if (this.isBatchUnlockLocked) {
+      this.context.logger.debug(
+        `batch unlock processing is locked for ${giveChain}`
+      );
+      return;
+    }
+
+    // execute batch unlock processing for full batch
+    if (orderIds.size >= this.params.batchUnlockSize) {
+      this.processUnlockBatches(giveChain, context);
+    }
+  }
+
+  private async processUnlockBatches(
+    giveChain: ChainId,
+    context: OrderProcessorContext
+  ) {
+    const logger = this.context.logger.child({
+      func: "processUnlockBatches",
+      giveChain,
+    });
+    logger.info("Batch unlocking is started");
+
+    try {
```

Comment:
There are toooooooooo many lines in the `try{}` block. It should be far more limited scope, e.g.:

```
fn performBatchUnlock(chainId) {
  if (lock) return;

  lock = true;
  const batchOfOrderIds = batch.slice(0, 10);
  res = tryUnlockBatch(batch);
  lock = false;

  if (res) {
    if (has_batches_in_any_chain)
     performBatchUnlock()
  }
}

fn tryUnlockBatch() {
if (Solana) {
  for (order in orders) {
    prepareTx;
    try {
      sendTransaction;
      removeOrderFromBatch;
    } catch (e) {  }
  }
}
else {
  prepareTx;
  try {
    sendTx;
    removeOrdersFrombatch()
  } catch (e) { return false; }
}
}


```

Reply:
Unnecessary try-catch